src/keyboardNavigation.ts

```diff
@@ -0,0 +1,83 @@
+import { isDeepestVisible, isImmediateParent } from './utils';
+
+function findDeepestVisibleChild(node: HTMLElement): HTMLElement {
+    if (isDeepestVisible(node)) return node;
+    const ul = <HTMLElement>node.children[1];
+    return findDeepestVisibleChild(<HTMLElement>ul.children[ul.children.length - 1]);
+}
+
+function findNearestParentWithAdditionalLeaves(node: HTMLElement): HTMLElement {
+    const ul = node.parentElement;
```

Comment:
some spots like this where it's a little code smelly to take a "node" but then know that the parent is specifically an unordered list